pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -802,6 +803,11 @@ protected boolean commitSegment(String controllerVipUrl, boolean isSplitCommit)
   }
 
   protected boolean buildSegmentAndReplace() {
+    if (!(closePartitionLevelConsumer() && closeStreamMetadataProvider())) {
+      return false;
```

Comment:
We should not stop the world if we cannot close the consumer.  If for any reason, the close fails, we should be able to continue, giving the gurantee that we will not call the consume methods on the same consumer. I am not opposed to changing the interface to support an explicit release() method (instead of just be auto-closeable). @npawar  can also comment on this.

Reply:
If the Kafka consumer cannot be closed, it's ok to just return false. 
Once `buildSegmentAndReplace()` gets called, it indicates that we don't need Kafka consumer any more. 
Thus, returning false and then downloading another one should be fine.

Reply:
I think this could result in no segment being built. My guess is that is we get an exception when closing the partitionLevelConsumer or the streamMetadataConsumer, it'll likely be a stream issue, and we'll get exceptions across all partitions, and no one will build the segment. 
Wdyt?

Reply:
The current way we handle the exception when the consumer cannot be closed is that we just swallow the exception and log an error message.
We can still swallow the message in this new PR.

Reply:
@npawar Updated the PR to swallow the exception.